src/sentry/api/event_search.py

```diff
@@ -1264,16 +1264,31 @@ def get_filter(query=None, params=None):
 }
 
 
-def get_json_meta_type(field_alias, snuba_type):
+def get_json_meta_type(field_alias, snuba_type, field=None):
     alias_definition = FIELD_ALIASES.get(field_alias)
     if alias_definition and alias_definition.get("result_type"):
         return alias_definition.get("result_type")
+
     snuba_json = get_json_type(snuba_type)
-    function_match = FUNCTION_ALIAS_PATTERN.match(field_alias)
-    if function_match and snuba_json != "string":
-        function_definition = FUNCTIONS.get(function_match.group(1))
-        if function_definition and function_definition.result_type:
-            return function_definition.result_type
+    if snuba_json != "string":
+        if field is not None:
+            function_match = FUNCTION_PATTERN.search(field)
```

Comment:
Would be nice if there was a way to pass in the Function objects here so that we didn't need to duplicate the parsing work.

Reply:
Do you think we could fit/move this logic into the `snuba.discover.transform_results`? Or do we not have enough context there either?

Reply:
Unfortunately, to avoid duplicating the parsing work, we'd have to lift all of that out from `resolve_field_list > resolve_field > resolve_function` and pass the results around. This will require a lot more work to accomplish as these functions are used in other places that will need to be updated.

Reply:
Oof that is a big chunk of work. Lets punt on that for now.

Reply:
This PR has been refactored and the parsing is no longer duplicated.